src/Symfony/Component/Form/Form.php

```diff
@@ -1044,11 +1044,29 @@ public function createView(FormView $parent = null)
             $view->children[$name] = $child->createView($view);
         }
 
+        $this->sort($view->children);
+
         $type->finishView($view, $this, $options);
 
         return $view;
     }
 
+    /**
+     * Sorts view fields based on their priority value.
+     */
+    private function sort(array &$children): void
+    {
+        $c = [];
+        $i = 0;
+        foreach ($children as $name => $child) {
+            $c[$name] = ['p' => $child->vars['priority'], 'i' => $i++];
+        }
+
+        uksort($children, static function ($a, $b) use ($c): int {
```

Comment:
maybe we should skip this call if the previous foreach detects that all priorities are `0`?

Reply:
Better, updated.